icgcget/commands/download.py

```diff
@@ -29,132 +30,130 @@
         self.logger = logging.getLogger("__log__")
-        self.gdc_client = GdcDownloadClient(pickle_path)
-        self.ega_client = EgaDownloadClient(pickle_path)
-        self.gt_client = GnosDownloadClient(pickle_path)
-        self.pdc_client = PdcDownloadClient(pickle_path)
-        self.icgc_client = StorageClient(pickle_path)
+        self.gdc_client = GdcDownloadClient(pickle_path, docker)
+        self.ega_client = EgaDownloadClient(pickle_path, docker)
+        self.gt_client = GnosDownloadClient(pickle_path, docker)
+        self.pdc_client = PdcDownloadClient(pickle_path, docker)
+        self.icgc_client = StorageClient(pickle_path, docker)
 
-    def download_manifest(self, repos, file_ids, manifest, output, api_url, verify):
+    def download_manifest(self, repos, file_ids, manifest, output, api_url, verify, unique=False):
         portal = portal_client.IcgcPortalClient(verify)
         manifest_json = self.get_manifest(manifest, file_ids, api_url, repos, portal)
-        session_info = {'pid': os.getpid(), 'start_time': datetime.datetime.utcnow().isoformat(), 'command': file_ids}
-        size = calculate_size(manifest_json, session_info)
-        object_ids = session_info['object_ids']
+        download_session = {'pid': os.getpid(), 'start_time': datetime.datetime.utcnow().isoformat(),
+                            'command': file_ids}
+        size, download_session = calculate_size(manifest_json, download_session)
+        file_data = download_session['file_data']
         if manifest:
             file_ids = []
-            for repo in object_ids:
-                file_ids.extend(object_ids[repo].keys())
+            for repo in file_data:
+                file_ids.extend(file_data[repo].keys())
 
         entities = api_error_catch(self, portal.get_metadata_bulk, file_ids, api_url)
         for entity in entities:
-            for repo_id in object_ids:
-                if entity['id'] in object_ids[repo_id]:
-                    repo = repo_id
-                    break
-            else:
-                self.logger.warning("File %s not found on any of the specified repositories", entity["id"])
-                continue
-
-            file_copies = entity['fileCopies']
-            for copy in file_copies:
-                if copy['repoCode'] == repo:
-                    if search_recursive(copy["fileName"], output):
-                        object_ids[repo].pop(entity['id'])
+            repo, copy = match_repositories(self, repos, entity)
+            if not repo:
+                raise click.Abort()
+
+            if copy['repoCode'] == repo:
+                if unique and search_recursive(copy["fileName"], output):
+                    file_data[repo].pop(entity['id'])
+                    self.logger.warning("File %s found in download directory, skipping", entity['id'])
+                    continue
+                file_data[repo][entity["id"]]['fileName'] = copy["fileName"]
```

Comment:
Please be dry here:

```python
file = file_data[repo][entity["id"]]
```

Will cut down a lot of noise. Please use this technique going forward.